pack.js

```diff
@@ -490,6 +490,11 @@ export class Packr extends Unpackr {
 						} else {
 							if (value.toJSON) // use this as an alternate mechanism for expressing how to serialize
 								return pack(value.toJSON());
+							
+							// if there is a writeFunction, use it, otherwise just encode as undefined
+							if (constructor === Function)
```

Comment:
I believe that one case that this misses is packing a generator function `{*generatorFunction(){}}` (and async generator, which also has a distinct constructor). I think we could just have this line use the `(type === 'function')` condition?